shared/wallets/transaction/index.js

```diff
@@ -0,0 +1,228 @@
+      ifFlowErrorsHereItsCauseYouDidntHandleAllActionTypesAbove(props.counterpartyType);
+      */
+      break
+  }
+
+  const amount = <Text type={textTypeSemibold}>{props.amountUser}</Text>
+
+  if (props.counterpartyType === 'wallet') {
+    const verbPhrase = props.pending ? 'Transferring' : 'You transferred'
+    if (props.yourRole === 'sender') {
+      return (
+        <Text type={textType}>
+          {verbPhrase} Lumens worth {amount} from this wallet to {counterparty}.
+        </Text>
+      )
+    }
+
+    return (
+      <Text type={textType}>
+        {verbPhrase} Lumens worth {amount} from {counterparty} to this wallet.
+      </Text>
+    )
+  }
+
+  if (props.yourRole === 'sender') {
+    const verbPhrase = props.pending ? 'Sending' : 'You sent'
+    return (
+      <Text type={textType}>
+        {verbPhrase} Lumens worth {amount} to {counterparty}.
+      </Text>
+    )
+  }
+
+  const verbPhrase = props.pending ? 'sending' : 'sent you'
+  return (
+    <Text type={textType}>
+      {counterparty} {verbPhrase} Lumens worth {amount}.
+    </Text>
+  )
+}
+
+type AmountXLMProps = {|
+  yourRole: Role,
+  amountXLM: string,
+  pending: boolean,
+|}
+
+const AmountXLM = (props: AmountXLMProps) => {
+  const color = props.pending
+    ? globalColors.black_20
+    : props.yourRole === 'sender'
+      ? globalColors.red
+      : globalColors.green
+  const amount = `${props.yourRole === 'sender' ? '-' : '+'} ${props.amountXLM}`
+  return (
+    <Text style={{color, textAlign: 'right'}} type="BodyExtrabold">
+      {amount}
+    </Text>
+  )
+}
+
+type TimestampProps = {|
+  timestamp: Date | null,
+|}
+
+const Timestamp = (props: TimestampProps) => {
+  if (!props.timestamp) {
+    return <Text type="BodySmall">Pending</Text>
+  }
+  const m = moment(props.timestamp)
+  return (
+    <Text title={m.format()} type="BodySmall">
+      {m.calendar()}
+    </Text>
+  )
+}
+
+export type Props = {|
+  large: boolean,
+  // A null timestamp means the transaction is still pending.
+  timestamp: Date | null,
+  yourRole: Role,
+  counterparty: string,
+  counterpartyType: CounterpartyType,
+  amountUser: string,
+  amountXLM: string,
+  note: string,
+|}
+
+export const Transaction = (props: Props) => {
+  const pending = !props.timestamp
+  return (
+    <Box2 direction="horizontal" fullWidth={true} style={styles.container}>
+      <CounterpartyIcon
+        counterparty={props.counterparty}
+        counterpartyType={props.counterpartyType}
+        large={props.large}
+      />
+      <Box2 direction="vertical" fullHeight={true} style={styles.rightContainer}>
+        <Timestamp timestamp={props.timestamp} />
+        <Detail
+          large={props.large}
+          pending={pending}
+          yourRole={props.yourRole}
+          counterparty={props.counterparty}
+          counterpartyType={props.counterpartyType}
+          amountUser={props.amountUser}
+        />
+        {// TODO: Detect single emoji case and use big emoji.
```

Comment:
This isn't easy without going through the markdown parser, although I'm still reluctant to go down that route. Let's talk about it tomorrow...

Reply:
We should definitely use markdown for these details

Reply:
When I did this in the chat/conversation/wallet payments, I injected the gray 'quote' bar manually to the left of the markown to preserve that formatting.

Reply:
@chrisnojima ? I think all my points from the comment above still apply:

A few reasons:

1) The notes are a stellar thing that's not inherently markdown, right? Seems wrong to assume they are.
2) Just because it's safe in chat doesn't mean it's safe in notes from rando Stellar addresses. For example, I don't think we want to linkify anything in a note, do we?
3) Our current method of quoting doesn't always work given ``` blocks, etc. (we already have to do this in a different way)

I think we can make single emojis big without markdown.

Reply:
Ah, okay, that makes sense thanks

Reply:
Fixed now